internal/innosat/reader.go

```diff
@@ -0,0 +1,58 @@
+package innosat
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+
+	"github.com/howeyc/crc16"
+)
+
+// SourcePayload ...
+type SourcePayload interface {
+	PUSVersion() uint8
+}
+
+// SourcePackage ...
```

Comment:
Since this one has the strange `Application` perhaps some info on what that is compared to `Payload`.